testing/fake-dom.js

```diff
@@ -98,6 +98,35 @@ export class FakeWindow {
       },
     });
 
+    /** @private {!Array<string>} */
+    this.cookie_ = [];
+    Object.defineProperty(this.document, 'cookie', {
+      get: () => {
+        let cookie = [];
+        for (let i = 0; i < this.cookie_.length; i += 2) {
+          cookie.push(`${this.cookie_[i]}=${this.cookie_[i + 1]}`);
+        }
+        return cookie.join(';');
+      },
+      set: value => {
+        const semi = value.indexOf(';');
+        const cookie = value.match(/^([^=]*)=([^;]*)/);
+        const expiresMatch = value.match(/expires=([^;]*)(;|$)/);
+        const expires = expiresMatch ? Date.parse(expiresMatch[1]) : Infinity;
+        let i = 0;
+        for (; i < this.cookie_.length; i += 2) {
+          if (this.cookie_[i] == cookie[1]) {
+            break;
+          }
+        }
+        if (Date.now() >= expires) {
```

Comment:
Ideally, but only real use of expires I'm aware of is to delete the cookie immediately. So this worked well.